test/async-route-handler.test.js

```diff
@@ -36,15 +35,14 @@ target.listen({ port: 0 }, (err) => {
     base: `http://localhost:${target.address().port}`
   })
 
-  instance.listen({ port: 0 }, (err) => {
+  instance.listen({ port: 0 }, async (err) => {
```

Comment:
The `listen` method accepts a sync function. Providing an `async` one should be avoided because if someone reads our test learns something that should not be done (even tho it works)

Reply:
+1, it would be better to switch to async/await completely in the test files.

Reply:
You can use the event.once utility combined with the listening event emitted from the server.

Reply:
Moving them to async/await is my next goal, but it is too much stuff to put in a single PR imho